pkg/input/arm/eval.go

```diff
@@ -22,23 +22,20 @@ import (
 // references to implementations of functions, and the other fields exist to
 // pass data to certain function impls other than what is already in their args.
 type EvaluationContext struct {
-	discoveredResourceSet map[string]struct{}
-	variables             map[string]interface{}
-	funcs                 map[string]armFnImpl
+	DiscoveredResourceSet map[string]struct{}
+	Variables             map[string]interface{}
+	BuiltinFunctions      map[string]Function
```

Comment:
```suggestion
	Functions      map[string]Function
```

If the idea is for callers to extend the function set, wouldn't it just be Functions?

But also, why is it desirable for callers to be able to extend the function set? Is this package not really just an internal dependency of pkg/input?

Reply:
I did the rename.

I wasn't necessarily thinking about extending it; more the other way around: restrict functions that aren't available in certain passes or replace them by dummies.

Reply:
Ah I think I see. Do you mean that in the parent package, in arm.go, we might in a multi-pass future add functions to the context that can now be meaningfully evaluated after passes? That makes sense if so.